File: 0066-Plus-One/0066.py
from typing import List

class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        idx = len(digits) - 1

        # iterate from left to right
        while idx >= 0:
            sum = digits[idx] + 1
            if sum < 10:
                digits[idx] = sum
                return digits
            else:
                digits[idx] = 0
                idx -= 1

        # means the first number of original array is greater than 9, therefor insert value 1 in 0 index
        digits.insert(0, 1)

        return digits

File: 0066-Plus-One/0066.java
class Solution {
    public int[] plusOne(int[] digits) {
        int idx = digits.length - 1;

        // iterate from right to left
        for (int i = idx; i >= 0; i--) {
            if (digits[i] < 9) {
                digits[i]++;
                return digits;
            }
            digits[i] = 0;
        }

        // [9, 9, 9] -> [0, 0, 0, 0]
        digits = new int[digits.length + 1];

        // [0, 0, 0, 0] -> [1, 0, 0, 0]
        digits[0] = 1;
        return digits;
    }
}